tdrs-backend/tdpservice/reports/views.py

```diff
@@ -21,11 +21,35 @@ class ReportFileViewSet(ModelViewSet):
     """Report file views."""
 
     http_method_names = ['get', 'post', 'head']
+    filterset_fields = ['year', 'quarter']
     parser_classes = [MultiPartParser]
     permission_classes = [ReportFilePermissions]
     serializer_class = ReportFileSerializer
+
+    # TODO: Handle versioning in queryset
+    # Ref: https://github.com/raft-tech/TANF-app/issues/1007
     queryset = ReportFile.objects.all()
 
+    # NOTE: This is a temporary hack to make sure the latest version of the file
+    # is the one presented in the UI. Once we implement the above linked issue
+    # we will be able to appropriately refer to the latest versions only.
+    ordering = ['-version']
+
+    def get_queryset(self):
+        """Determine the queryset used to fetch records for users."""
+        user = self.request.user
+
+        # OFA Admins can see reports for all STTs
+        if is_in_group(user, 'OFA Admin'):
```

Comment:
Note: This filtering is needed in addition to what is done in `ReportFilePermissions.has_object_permission` since that function does not get called for list endpoints, only when retrieving a single report. With #888 we could improve this using more abstracted object permissions.